file/random_access_file_reader.cc

```diff
@@ -359,6 +356,35 @@ IOStatus RandomAccessFileReader::MultiRead(
 
     {
       IOSTATS_CPU_TIMER_GUARD(cpu_read_nanos, clock_);
+      if (rate_limiter_priority != Env::IO_TOTAL && rate_limiter_ != nullptr) {
+        assert(fs_reqs != nullptr);
+        // TODO: ideally we should request bandwith for each
+        // `FSReadRequest::len` seperately instead of requesting
+        // the sum and calling `LoopRateLimitRequestHelper()` afterwards
+        size_t total_multi_read_size = 0;
+        for (size_t i = 0; i < num_fs_reqs; ++i) {
+          FSReadRequest& req = fs_reqs[i];
+          total_multi_read_size += req.len;
+        }
+        static auto LoopRateLimitRequestHelper =
+            [](const size_t total_bytes_to_request, RateLimiter* rate_limiter,
+               const Env::IOPriority pri, Statistics* stats,
+               const RateLimiter::OpType op_type) {
+              assert(rate_limiter != nullptr);
+              size_t remaining_bytes = total_bytes_to_request;
+              size_t request_bytes = 0;
+              while (remaining_bytes > 0) {
+                request_bytes = std::min(
+                    static_cast<size_t>(rate_limiter->GetSingleBurstBytes()),
+                    remaining_bytes);
+                rate_limiter->Request(request_bytes, pri, stats, op_type);
+                remaining_bytes -= request_bytes;
+              }
```

Comment:
This doesn't need to be executed via a function object, does it?

Reply:
Hmmm - I think it's still thru function object under the hood. Actually I wanted to reach out for help on this and the previous PR is just a WIP.  

My goal is to encapsulate the helper function `LoopRateLimitRequestHelper` within this particular function `::MultiRead()` to minimize its usage as much as possible (as you can see from our plan of deprecating `BackupEngineImpl::LoopRateLimitRequestHelper`). 

So I am trying something like local (so that it can't be called outside of this function) static (so that we only have one copy) [anonymous function](https://en.cppreference.com/w/cpp/language/lambda). According the website title, "Lambda expressions - Constructs a [closure](https://en.wikipedia.org/wiki/Closure_(computer_science)): an unnamed function object capable of capturing variables in scope.", I think a function object is still constructed.

I am not sure if this is the best practice to achieve my goal though so let me know if you see alternatives!

Reply:
I'd delete line 369-372 and line 383-386, and then fixup the undefined variables.

Reply:
Okay - so basically just remove the function abstraction - I can do that too.